src/runtime_src/core/common/api/xrt_module.cpp

```diff
@@ -1885,6 +1926,140 @@ class module_sram : public module_impl
     return payload;
   }
 
+  struct dlib_handle
+  {
+    void* handle = nullptr;
+
+    dlib_handle(const std::string& path)
+    {
+      handle = xrt_core::dlopen(path.c_str(), RTLD_LAZY);
+    }
+
+    ~dlib_handle()
+    {
+      if (handle)
+        xrt_core::dlclose(handle);
+    }
+  };
+
+  struct dtrace_util
+  {
+    std::unique_ptr<dlib_handle> lib_hdl;
+    std::string ctrl_file_path;
+    std::string map_data;
+  };
+
+  static bool
+  init_dtrace_helper(const module_impl* parent, dtrace_util& dtrace_obj)
+  {
+    // The APIs used for dynamic tracing in future will be checked into
+    // AIEBU repo after Telluride related code is made public.
+    // Till then we are using local library whose path is specified using
+    // ini option. Instead of having the library as a dependency to XRT we
+    // are using dlopen/dlsym methods.
+    // TODO : remove ini options and dlopen/dlsym logic after the dtrace
+    //        API's are integrated into AIEBU repo
+    static auto dtrace_lib_path = xrt_core::config::get_dtrace_lib_path();
+    if (dtrace_lib_path.empty())
+      return false; // dtrace lib path is not set, dtrace is not enabled
+
+    dtrace_obj.lib_hdl = std::make_unique<dlib_handle>(dtrace_lib_path);
+    if (!dtrace_obj.lib_hdl->handle) {
+      xrt_core::message::send(xrt_core::message::severity_level::debug, "xrt_module",
+                              "Failed to load dtrace library");
+      return false;
+    }
+
+    static auto path = xrt_core::config::get_dtrace_control_file_path();
+    if (!std::filesystem::exists(path)) {
+      xrt_core::message::send(xrt_core::message::severity_level::debug, "xrt_module",
+                              "Dtrace control file is not accessible");
+      return false;
+    }
+    dtrace_obj.ctrl_file_path = path;
+
+    static const auto& data = parent->get_dump_buf();
+    if (data.size() == 0) {
+      xrt_core::message::send(xrt_core::message::severity_level::debug, "xrt_module",
+                              "Dump section is empty in ELF");
+      return false;
+    }
+
+    dtrace_obj.map_data = std::string{reinterpret_cast<const char*>(data.data()), data.size()};
+    return true;
+  }
+
+  void
+  initialize_dtrace_buf(const module_impl* parent)
+  {
```

Comment:
Since this function is rather long, can we insert some comments in between the various blocks as to what exactly is being initialized?